conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java

```diff
@@ -186,17 +199,33 @@ private static void error(HttpServerExchange exchange, Error error) {
         writeResponse(exchange, Optional.empty(), ErrorType.INTERNAL.httpErrorCode());
     }
 
+    private static void writeErrorResponse(HttpServerExchange exchange, ConjureError error, int statusCode) {
+        writeResponseInternal(exchange, statusCode, () -> errorSerializer.serialize(error, exchange));
+    }
+
     private static void writeResponse(
             HttpServerExchange exchange, Optional<SerializableError> maybeBody, int statusCode) {
+        writeResponseInternal(exchange, statusCode, () -> {
+            if (maybeBody.isPresent()) {
+                serializer.serialize(maybeBody.get(), exchange);
+            }
+        });
+    }
+
+    @FunctionalInterface
+    private interface SerializerRunnable {
+        void run() throws IOException;
+    }
```

Comment:
not 100% on it at this point, but I'm curious what happens to the diff if we add `static ConjureError fromServiceException(ServiceException exception) {` to `ConjureError` and use the new type for all cases in this file? The new factory would need to implement the old behavior which calls `Objects.toString` on all inputs, to avoid making wire changes.

Reply:
This works. Removed the use of the slightly hacky `SerializerRunnable`. I also added a `fromRemoteException` version.